File: android-kit-base/src/test/java/com/mparticle/kits/KitUtilsTest.java
package com.mparticle.kits;

import org.junit.Test;

import java.util.LinkedList;
import java.util.List;

import static org.junit.Assert.*;

public class KitUtilsTest {

    @Test
    public void testSanitizeAttributeKey() throws Exception {
        assertNull(KitUtils.sanitizeAttributeKey(null));
        assertEquals("", KitUtils.sanitizeAttributeKey(""));
        assertEquals("TestTest", KitUtils.sanitizeAttributeKey("$TestTest"));
        assertEquals("$", KitUtils.sanitizeAttributeKey("$$"));
        assertEquals("", KitUtils.sanitizeAttributeKey("$"));
    }

    @Test
    public void testJoin() throws Exception {
        List<String> testList = new LinkedList<>();
        testList.add("1");
        testList.add("test");
        testList.add("test 2");
        assertEquals("1,test,test 2", KitUtils.join(testList));
        assertEquals("", KitUtils.join(new LinkedList<String>()));
        List<String> singleElementList = new LinkedList<>();
        singleElementList.add("tester");
        assertEquals("tester", KitUtils.join(singleElementList));
    }

    @Test
    public void testJoinWithDelimiter() throws Exception {
        List<String> testList = new LinkedList<>();
        testList.add("1");
        testList.add("test");
        testList.add("test 2");
        assertEquals("", KitUtils.join(new LinkedList<String>(), "whatever"));
        List<String> singleElementList = new LinkedList<>();
        singleElementList.add("tester");
        assertEquals("tester", KitUtils.join(singleElementList, "whatever"));
        assertEquals("1whatevertestwhatevertest 2", KitUtils.join(testList, "whatever"));
    }
}

File: android-core/src/test/kotlin/com/mparticle/kits/KitUtilsTest.kt
package com.mparticle.kits

import org.junit.Assert
import org.junit.Test
import java.util.LinkedList

class KitUtilsTest {
    @Test
    @Throws(Exception::class)
    fun testSanitizeAttributeKey() {
        Assert.assertNull(KitUtils.sanitizeAttributeKey(null))
        Assert.assertEquals("", KitUtils.sanitizeAttributeKey(""))
        Assert.assertEquals("TestTest", KitUtils.sanitizeAttributeKey("\$TestTest"))
        Assert.assertEquals("$", KitUtils.sanitizeAttributeKey("$$"))
        Assert.assertEquals("", KitUtils.sanitizeAttributeKey("$"))
    }

    @Test
    @Throws(Exception::class)
    fun testJoin() {
        val testList: MutableList<String> = LinkedList()
        testList.add("1")
        testList.add("test")
        testList.add("test 2")
        Assert.assertEquals("1,test,test 2", KitUtils.join(testList))
        Assert.assertEquals("", KitUtils.join(LinkedList()))
        val singleElementList: MutableList<String> = LinkedList()
        singleElementList.add("tester")
        Assert.assertEquals("tester", KitUtils.join(singleElementList))
    }

    @Test
    @Throws(Exception::class)
    fun testJoinWithDelimiter() {
        val testList: MutableList<String> = LinkedList()
        testList.add("1")
        testList.add("test")
        testList.add("test 2")
        Assert.assertEquals("", KitUtils.join(LinkedList(), "whatever"))
        val singleElementList: MutableList<String> = LinkedList()
        singleElementList.add("tester")
        Assert.assertEquals("tester", KitUtils.join(singleElementList, "whatever"))
        Assert.assertEquals("1whatevertestwhatevertest 2", KitUtils.join(testList, "whatever"))
    }
}
